FreeRTOS/Demo/CORTEX_MPU_M4_MPS2_QEMU_GCC/Makefile

```diff
@@ -0,0 +1,87 @@
+CC                     =    arm-none-eabi-gcc
+BIN                   :=    RTOSDemo.axf
+
+BUILD_DIR             :=    build
+
+FREERTOS_DIR_REL      :=    ../../../FreeRTOS
+FREERTOS_DIR          :=    $(abspath $(FREERTOS_DIR_REL))
+KERNEL_DIR            :=    $(FREERTOS_DIR)/Source
+
+FREERTOS_PLUS_DIR_REL :=    ../../../FreeRTOS-Plus
+FREERTOS_PLUS_DIR     :=    $(abspath $(FREERTOS_PLUS_DIR_REL))
+
+SOURCE_FILES          +=    $(KERNEL_DIR)/portable/GCC/ARM_CM4_MPU/port.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/portable/Common/mpu_wrappers.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/tasks.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/list.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/queue.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/timers.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/event_groups.c
+SOURCE_FILES          +=    $(KERNEL_DIR)/stream_buffer.c
+SOURCE_FILES          +=    ${KERNEL_DIR}/portable/MemMang/heap_4.c
+
+INCLUDE_DIRS          +=    -I.
+INCLUDE_DIRS          +=    -I$(KERNEL_DIR)/include
+INCLUDE_DIRS          +=    -I$(KERNEL_DIR)/portable/GCC/ARM_CM4_MPU
+
+SOURCE_FILES          +=    main.c
+SOURCE_FILES          +=    app_main.c
+SOURCE_FILES          +=    mpu_demo.c
+SOURCE_FILES          +=    startup.c
+SOURCE_FILES          +=    syscall.c
+
+
+DEFINES               :=    -DQEMU_SOC_MPS2
+DEFINES               :=    -DHEAP4
+DEFINES               +=    -DportUSING_MPU_WRAPPERS=1
+
+CPPFLAGS              +=    $(DEFINES)
+
+CFLAGS                +=    -mthumb
+CFLAGS                +=    -mfloat-abi=softfp
```

Comment:
do we want soft or hard fp? Perhaps we should build a few different variants.

Reply:
the M3 is running with hard, 
i had some problems running this one with hard, so did it with soft